engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageState.java

```diff
@@ -180,6 +221,18 @@ public void put(final long key, final MessageRecord record) {
     deadline.wrapLong(record.getDeadline());
     deadlineColumnFamily.insert(deadlineMessageKey, DbNil.INSTANCE);
 
+    long count = 0L;
+    if (!messagesDeadlineCountColumnFamily.isEmpty()) {
+      count = messagesDeadlineCountColumnFamily.get(messagesDeadlineCountKey).getValue();
+      messagesDeadlineCount.wrapLong(count + 1L);
+      messagesDeadlineCountColumnFamily.update(messagesDeadlineCountKey, messagesDeadlineCount);
+    } else {
+      messagesDeadlineCount.wrapLong(count + 1L);
+      messagesDeadlineCountColumnFamily.insert(messagesDeadlineCountKey, messagesDeadlineCount);
```

Comment:
:x:
```suggestion
      messagesDeadlineCountColumnFamily.upsert(messagesDeadlineCountKey, messagesDeadlineCount);
```
insert will fail if the key already exists.